access/src/main/external/static/js/admin/src/MoveActionMonitor.js

```diff
@@ -0,0 +1,255 @@
+define('MoveActionMonitor', [ 'jquery'], function($) {
+
+	var defaultOptions = {
+		updateInterval : 5000
+	};
+	
+	function MoveActionMonitor(alertHandler, options) {
+		this.options = $.extend({}, defaultOptions, options);
+		
+		this.alerts = alertHandler;
+		this.moveData = {};
+	}
+	
+	function setDifference(A, B) {
+		var map = {}, C = [];
+
+		for (var i = B.length; i--; )
+			map[B[i]] = null;
+
+		for (var i = A.length; i--; ) {
+			if (!map.hasOwnProperty(A[i]))
+				C.push(A[i]);
+		}
+
+		return C;
+	}
+	
+	MoveActionMonitor.prototype.setResultList = function(resultList) {
+		this.resultList = resultList;
+	};
+	
+	MoveActionMonitor.prototype.activate = function() {
+		this.active = true;
+		
+		// Mark objects that were previously known to be moving
+		var self = this;
+		$.each(this.moveData, function(moveId, moveDetails) {
+			self.markMoving(moveDetails.pids);
+		});
+		
+		// Start refresh loop
+		this.update();
```

Comment:
I did at first, but switched to make sure that the next run of the function wouldn't start until the ajax call from the previous one had finished in case the server decided to take a really long time to respond